public/app/core/config.ts

```diff
@@ -1,5 +1,18 @@
 import { config, GrafanaBootConfig } from '@grafana/runtime';
-
 // Legacy binding paths
 export { config, GrafanaBootConfig as Settings };
+
+let configMock: Partial<GrafanaBootConfig> | null = null;
+
 export default config;
+
+export const getConfig = () => {
+  return configMock || config;
+};
+
+export const mockConfig = (mock: Partial<GrafanaBootConfig>) => {
+  configMock = mock;
+  return () => {
+    configMock = null;
+  };
```

Comment:
do we need the configMock object? Can't we have a setConfig function instead that set's the config variable?

Reply:
@torkelo mind the type. This allows partial config mocking, only what you need for your test. That's why I opted for "mock" naming

Reply:
can't it be just
```
export function updateConfig(updates: Partial<GrafanaBootConfig>) {
   Object.assign(config, mock);
}
```

Reply:
I like this!

Reply:
This function: 

```ts
export const getConfig = () => {
  return configMock || config;
};
```

feels wrong to have in prod code.

Reply:
@torkelo it's no longer there